store/sqlstore/thread_store.go

```diff
@@ -133,17 +137,57 @@ func (s *SqlThreadStore) GetThreadsForUser(userId string, opts model.GetUserThre
 	if opts.PageSize == 0 {
 		pageSize = opts.PageSize
 	}
+
+	mentionsQuery, mentionsQueryArgs, _ := s.getQueryBuilder().
```

Comment:
In this query, the mention count should be properly calculated before you reach this point, so the you don't need the join with the post table or check anything related to posts.